addons/website/controllers/main.py

```diff
@@ -464,13 +464,16 @@ def website_image(self, model=None, id=None, field=None, xmlid=None, max_width=N
         xmlid can be used to load the image. But the field image must by base64-encoded
         """
         if xmlid and "." in xmlid:
-            xmlid = xmlid.split(".", 1)
             try:
-                model, id = request.registry['ir.model.data'].get_object_reference(request.cr, request.uid, xmlid[0], xmlid[1])
+                record = request.env['ir.model.data'].xmlid_to_object(xmlid)
```

Comment:
If you're going to switch that to the new API, why not just use `request.env.ref(xmlid)`?